#!/usr/bin/env python
PACKAGE = "unicon"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()



gen.add("CtrlOmega_xy", double_t, 0, "A double parameter", 2.0, 0.0, 5.0)
gen.add("CtrlOmega_z", double_t, 0, "A double parameter", 2.8, 0.1, 5.0)
gen.add("CtrlZita_xy", double_t, 0, "A double parameter", 1.0, 0.1, 3.0)
gen.add("CtrlZita_z", double_t, 0, "A double parameter", 1.0, 0.1, 3.0)

gen.add("PosErrorAccumulatedLimit_xyz", double_t, 0, "A double parameter",10, 0.0, 30)
gen.add("Pos_ki_xy", double_t, 0, "A double parameter", 0.03, 0, 1.0)
gen.add("Pos_ki_z", double_t, 0, "A double parameter", 0.07, 0, 1.0)


gen.add("k_p_yaw_", double_t, 0, "A double parameter", 2.0, 0.1, 3.0)

gen.add("k_I_yaw_", double_t, 0, "A double parameter", 0.02, 0, 0.5)

gen.add("yawErrorAccumLim_", double_t, 0, "A double parameterr", 1.57, 0, 3.14)

gen.add("mpc_accel_weight_", double_t, 0, "A double parameterr", 7.0, 0, 10.0)
gen.add("mpc_vel_weight_", double_t, 0, "A double parameterr", 0.0, 0, 2.0)

gen.add("thrust_", double_t, 0, "A double parameterr", 89.75, 10, 170)

gen.add("max_angle_", double_t, 0, "A double parameterr", 15, 0, 60)

gen.add("max_yaw_rate_", double_t, 0, "A double parameterr", 90, 0, 170)

gen.add("dji_thrust_coefficient", double_t, 0, "A double parameterr", 90, 0, 90)
gen.add("dji_minimum_thrust", double_t, 0, "A double parameter", 5, 0, 30)
gen.add("dji_maximum_thrust", double_t, 0, "A double parameter", 50, 40, 100)
gen.add("dji_thrust_offset", double_t, 0, "A double parameter", 1.5, 0, 100)

exit(gen.generate(PACKAGE, "unicon", "Param"))
